src/In.ProjectEKA.HipService/Discovery/PatientDiscovery.cs

```diff
@@ -56,9 +59,22 @@ public class PatientDiscovery
                     .Map(async patient =>
                     {
                         await discoveryRequestRepository.Add(new Model.DiscoveryRequest(request.TransactionId,
-                            request.Patient.Id, patient.Identifier));
+                            request.Patient.Id, patient.Identifier,
+                            patient.MaskedIdentifier));
+                        var unlinkedAccounts = GetUnlinkedCareContexts(linkedCareContexts, patient);
+                        var careContextRepresentations = unlinkedAccounts.ToList();
+                        var addToUnlinkedAccounts = AddToUnlinkedAccounts(careContextRepresentations, patient.Identifier);
+                        if (!careContextRepresentations.Any() &&
+                            !addToUnlinkedAccounts)
+                        {
+                            return (null,
+                                new ErrorRepresentation(new Error(ErrorCode.FailedToGetLinkedCareContexts,
+                                    "Unable to update unlinked care contexts")));
+                        }
+
+                        patient.Identifier = patient.MaskedIdentifier;

```

Comment:
why mutating this?

Reply:
In the discovery response, masked patient identifier is sent back

Reply:
Same as what I gave for zafar.